### PR TITLE
Embedded js: remove some dependency from jquery

### PR DESCRIPTION
**Pull request by stsewd**
This needs https://github.com/readthedocs/common/pull/155.

Basically replacing ajax calls with the fetch API, and using the DOM API for the manipulation of elements.

Note: we need to create a rule on cloudflare to not cache the analytics endpoint,
we were previously using a hack for this (passing a random query arg). In the future, we should set this at the view level (no-cache). Already did this for .org and .com.

### CODE REVIEW

File: readthedocs/core/static-src/core/js/doc-embed/search.js [hunk 1]
```diff
@@ -275,23 +296,27 @@ function attach_elastic_search_query_mkdocs(data) {
                 var results = data.results || [];
 
                 if (results.length) {
-                    var searchResults = $('#mkdocs-search-results');
-                    searchResults.empty();
+                    let searchResults = document.getElementById('mkdocs-search-results');
```
Comment (stsewd) [inline readthedocs/core/static-src/core/js/doc-embed/search.js:299]: We could also just kill the mkdocs integration, isn't currently being used (only on my test project).

Comment (humitos) [reply readthedocs/core/static-src/core/js/doc-embed/search.js:299]: I'm 👍🏼 on removing things we are not using 😄

Comment (agjohnson) [reply readthedocs/core/static-src/core/js/doc-embed/search.js:299]: Perhaps this is something that we instead address with generic version of our search integration. I'm :+1: in both directions, if we aren't using the code it shouldn't hurt to keep it around either, but also happy to see unused code disappeared.

**Review by agjohnson — COMMENTED** [review]
These changes look good! It's hard to verify without a proper test suite, but this is something to keep on our radar.

File: readthedocs/core/static-src/core/js/doc-embed/search.js [hunk 2]
```diff
@@ -58,19 +75,19 @@ function attach_elastic_search_query_sphinx(data) {
...
             var div_title = document.createElement("div");
             var a_element = document.createElement("a");
             a_element.href = link;
             a_element.innerHTML = title;
             div_title.appendChild(a_element);
-            html = div_title.outerHTML;
-            for (var i = 0; i < contents.length; i += 1) {
-                var div_content = document.createElement("div");
+            let elements = [div_title];
+            for (let i = 0; i < contents.length; i += 1) {
```
[the remark this answers is not in the document]
Comment (stsewd) [reply readthedocs/core/static-src/core/js/doc-embed/search.js]: Changed, a related question, any preference on using `forEach` or `for of` for arrays :thinking: ?

[the remark this answers is not in the document]
Comment (agjohnson) [reply readthedocs/core/static-src/core/js/doc-embed/search.js]: Good question. I'm not aware of any major differences, other than I have accidentally assumed an object was an Array instance, only to find out on the console that it is not, and does not have `forEach()`

File: readthedocs/core/static-src/core/js/doc-embed/search.js (hunk 1 above)
Comment (ericholscher) [reply readthedocs/core/static-src/core/js/doc-embed/search.js:299]: I think we should remove it, and work towards shipping our livesearch as a generic JS library. The less code we have to maintain the better. 👍

**Review by ericholscher — APPROVED** [review]
Looking at the previous comments, I think this is good to ship. @agjohnson if you want to give it another look, go ahead, but it seems like you already reviewed it and it's 👍 with the review updates.

**Review by agjohnson — APPROVED** [review]
Changes look good!

**Pull request merged**